REGRESSION(295182@main): Broke test-webkitpy

briannafan opened the pull request:
#### e6f56a695c32af5c0a404905150ce65f2bca8849
<pre>
REGRESSION(295182@main): Broke test-webkitpy
<a href="https://bugs.webkit.org/show_bug.cgi?id=293992">https://bugs.webkit.org/show_bug.cgi?id=293992</a>
<a href="https://rdar.apple.com/152536037">rdar://152536037</a>

Reviewed by Sam Sneddon.

Add mock repository to test_exporter unittests.
Find the WebKit repository using WebKitFinder and use
webkitscmpy mocks in upload unittests.

* Tools/Scripts/webkitpy/tool/commands/upload_unittest.py:
(UploadCommandsTest):
* Tools/Scripts/webkitpy/w3c/test_exporter.py:
(WebPlatformTestExporter.__init__):
* Tools/Scripts/webkitpy/w3c/test_exporter_unittest.py:
(TestExporterTest.test_export_invalid_token):
(TestExporterTest.test_export_wrong_token):
(TestExporterTest.test_has_wpt_changes):
(TestExporterTest.test_has_no_wpt_changes_for_no_diff):

Canonical link: <a href="https://commits.webkit.org/295829@main">https://commits.webkit.org/295829@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1d6dd801a5774db89b5b4a9aa9cd76964732a19d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106265 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26014 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16410 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111463 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56861 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108304 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26677 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34517 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80719 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109269 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21110 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95893 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61046 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/105742 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20632 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13992 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56301 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90458 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14027 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114323 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33403 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24629 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89791 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33767 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92122 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89492 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34363 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12177 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28987 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17221 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33328 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33074 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36427 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34672 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/Scripts/webkitpy/w3c/test_exporter.py:
```diff
@@ -63,7 +63,7 @@ def __init__(self, host, options, gitClass=Git, bugzillaClass=Bugzilla, WPTGitHu
 
         self._WPTGitHubClass = WPTGitHubClass
         self._gitClass = gitClass  # FIXME: Move git operations to webkitscmpy
-        self._repository = repository or local.Git(os.path.abspath(os.getcwd()))
```
Can we not just use WebKitFinder? Do something like https://github.com/WebKit/WebKit/blob/627fbaff8094b58ecbb5817511f77d5b91d96e2f/Tools/Scripts/webkitpy/common/net/bugzilla/test_expectation_updater.py#L232-L235?

gsnedders commented on Tools/Scripts/webkitpy/tool/commands/upload_unittest.py:
```diff
@@ -34,6 +34,7 @@
 from webkitpy.tool.commands.commandtest import CommandsTest
 from webkitpy.tool.commands.upload import *
 from webkitpy.tool.mocktool import MockOptions, MockTool
+from webkitscmpy import mocks as wmocks
```
Nit: I wonder if just importing `webkitcorepy.mocks` and `webkitscmpy.mocks` and being explicit throughout would be clearer? But definitely not worth blocking this on!

briannafan committed:
REGRESSION(295182@main): Broke test-webkitpy
https://bugs.webkit.org/show_bug.cgi?id=293992
rdar://152536037

Reviewed by Sam Sneddon.

Add mock repository to test_exporter unittests.
Find the WebKit repository using WebKitFinder and use
webkitscmpy mocks in upload unittests.

* Tools/Scripts/webkitpy/tool/commands/upload_unittest.py:
(UploadCommandsTest):
* Tools/Scripts/webkitpy/w3c/test_exporter.py:
(WebPlatformTestExporter.__init__):
* Tools/Scripts/webkitpy/w3c/test_exporter_unittest.py:
(TestExporterTest.test_export_invalid_token):
(TestExporterTest.test_export_wrong_token):
(TestExporterTest.test_has_wpt_changes):
(TestExporterTest.test_has_no_wpt_changes_for_no_diff):

Canonical link: https://commits.webkit.org/295829@main
